packages/utils/src/validation.ts

```diff
@@ -1,10 +1,17 @@
-export function isValidHttpUrl(urlStr: string): boolean {
-  let url;
+export function validHttpUrl(urlStr: string): URL | null {
```

Comment:
Returning `null` if the `urlStr` is invalid might not be obvious to the caller. Right now, it would also accept a URL starting with `file://...`